src/compile/data/formatparse.ts

```diff
@@ -68,7 +68,7 @@ export class ParseNode extends DataFlowNode {
       model.forEachFieldDef(fieldDef => {
         if (fieldDef.type === TEMPORAL) {
           parse[fieldDef.field] = 'date';
-        } else if (fieldDef.type === QUANTITATIVE) {
+        } else if (fieldDef.type === QUANTITATIVE || fieldDef.bin) {
```

Comment:
https://github.com/vega/vega-lite/pull/2922 has a better solution for this but the test is useful.